### DISCUSSION TITLE
Virtual Text and Snippets

### DISCUSSION THREAD

**Discussion by PhosCity** [LazyVim/LazyVim#54] [Q&A] [answered]
I am a beginner and use Lunarvim but wanted to give LazyVim a try. So this might be obvious but can someone tell me how to disable virtual text.

Also does LazyVim expose a folder for custom snippets where I can just add my snippets like Lunarvim or do I need to set it up all by myself.

**Comment by folke** [comment]
I'll add something to make it easier to configure diagnostics later today.

See also https://github.com/LazyVim/LazyVim/issues/55

**Comment by folke** [comment, marked as answer]
I just made an update to make it easier to change diagnostics.

You can just do:

```lua
  {
    "neovim/nvim-lspconfig",
    opts = {
      diagnostics = {
        virtual_text = false,
      },
    },
  },
```

**Reply by PhosCity** [reply]
Thank you so much!

**Reply by venkatesh2100** [reply]
![image](https://github.com/user-attachments/assets/bdb8dcc0-5930-4577-9bbd-196706df4c7c)
Yea Got it

**Comment by folke** [comment]
I need to check what LunarVim uses for snippets. But yes you need to properly configure LuaSnip yourself for this right now.

Haven't checked LunarVim, but this plugin seems to automatically load snippets from `lua/snippets` into luasnip https://github.com/molleweide/LuaSnip-snippets.nvim

**Reply by PhosCity** [reply]
It'd be awesome if LazyVim offered snippets support as well. But I've been reading the docs and trying to set up luasnip myself rn. Here's what I have till now:

```lua
return {
  {
    "L3MON4D3/LuaSnip",
    config = function()
      require("luasnip.loaders.from_lua").load({ paths = "~/.config/nvim/snippets" })
    end,
  },
}
```
Then I make lua.lua in the snippets folder for snippets for lua file and so on. It seems to be working.

**Reply by alexciarlillo** [reply]
> It'd be awesome if LazyVim offered snippets support as well. But I've been reading the docs and trying to set up luasnip myself rn. Here's what I have till now:
> 
> ```lua
> return {
>   {
>     "L3MON4D3/LuaSnip",
>     config = function()
>       require("luasnip.loaders.from_lua").load({ paths = "~/.config/nvim/snippets" })
>     end,
>   },
> }
> ```
> 
> Then I make lua.lua in the snippets folder for snippets for lua file and so on. It seems to be working.

Do you recall if you had to do anything else to get this to work? I am attempting it and still only appearing to get the default LazyVim config for LuaSnip.

I made a new plugin file, `snippets.lua` and added the following:

```
return {
  {
    "L3MON4D3/LuaSnip",
    config = function()
      require("luasnip.loaders.from_vscode").load({ paths = "~/.config/nvim/snippets" })
    end,
  },
}
```
(I also tried an alternate syntax I saw on the LuaSnip repo like: `...load({ paths = { "~/.config/nvim/snippets" } })` and I also tried `lazy_load`)

And in the snippets directory I have a simple test javascript snippet in a file called `javascript.json`:
```
{
  "a2am": {
    "scope": "javascript,typescript,javascriptreact,typescriptreact",
    "prefix": "a2am",
    "body": "\nObjectTools.ArrayToArrayMap(${1:array}, ${2:mapKey}, ${3:valueKey})\n"
  }
}
```

In a .JS file, when I type `a` I only get the snippet from friendly-snippets (I also confirmed the file is correctly detected as `javascript` from LSP info):
![image](https://user-images.githubusercontent.com/656039/221465715-689c9cd9-441c-443c-8695-fc31058c4b3b.png)

When I check the LazyVim profile I can see my plugin file is being loaded:
![image](https://user-images.githubusercontent.com/656039/221465818-916373f6-f972-4f90-bdc5-9788845c4868.png)


But at the bottom I also see `LuaSnip` is loading with `friendly-snippets`:
![image](https://user-images.githubusercontent.com/656039/221465916-ad1216c5-2f6f-48bc-9096-539b3f373f57.png)

This makes me think my plugin file settings are not overriding LuaSnip settings because if it did, it should not load `friendly-snippets` if I understand correctly.

Just looking for any possible ideas on why this wouldn't work for me. I am just trying out LazyVim for the first time today so there are no other configuration changes which might conflict.

**Reply by PhosCity** [reply]
@alexciarlillo Since I use lua snippets, I can't be sure but I think you also need a file named package.json in your snippets folder if you want to use vs_code snippets. Try creating the following:
```json
{
  "contributes": {
    "snippets": [
      {
        "language": "javascript",
        "path": "./javascript.json"
      },
    ]
  }
}
```
[See here](https://github.com/L3MON4D3/LuaSnip/blob/master/DOC.md#vscode) for more.

**Reply by alexciarlillo** [reply]
Thank you! That is indeed what I was missing 👍

**Comment by smokeyhallow** [comment]
Snippets works for me, but I can't figure out where to add new snippets.

**Comment by dimas-cyriaco** [comment]
I was able to make it work with this config (inside `.config/nvim/lua/plugins/snippets.lua`):

```lua
return {
  {
    "l3mon4d3/luasnip",
    config = function()
      require("luasnip.loaders.from_lua").load({ paths = "~/.config/nvim/snippets" })
    end,
  },
}
```

Then, inside `.config/nvim/snippets/lua.lua` I put the snippets for Lua:

```lua
-- https://www.ejmastnak.com/tutorials/vim-latex/luasnip/

return {
  -- A snippet that expands the trigger "hi" into the string "Hello, world!".
  require("luasnip").snippet({ trig = "hi" }, { t("Hello, world!") }),

  -- To return multiple snippets, use one `return` statement per snippet file
  -- and return a table of Lua snippets.
  require("luasnip").snippet({ trig = "foo" }, { t("Another snippet.") }),
}
```

Seems to be working fine.
